presto-cli/src/main/java/com/facebook/presto/cli/TsvPrinter.java

```diff
@@ -55,6 +55,13 @@ public void printRows(List<List<?>> rows, boolean complete)
     public void finish()
             throws IOException
     {
+        flush();
+    }
+
+    @Override
+    public void flush()
+            throws IOException
+    {
         printRows(ImmutableList.<List<?>>of(), true);
```

Comment:
[concern/question]
For me it is hacky. Everyone knows what is the responsibility of flush method in classes like Printer. Having there a code which additionally print something, breaks this responsibility contract. I'd prefer to printRows in finish method, but I don't know if it is doable.